airflow/providers/pinecone/hooks/pinecone.py

```diff
@@ -64,43 +69,80 @@ def get_ui_field_behaviour(cls) -> dict[str, Any]:
         """Return custom field behaviour."""
         return {
             "hidden_fields": ["port", "schema"],
-            "relabeling": {"login": "Pinecone Environment", "password": "Pinecone API key"},
+            "relabeling": {
+                "login": "Pinecone Environment",
+                "host": "Pinecone Host",
+                "password": "Pinecone API key",
+            },
         }
 
-    def __init__(self, conn_id: str = default_conn_name) -> None:
+    def __init__(
+        self,
+        conn_id: str = default_conn_name,
+        environment: str | None = None,
+        region: str | None = None,
+        api_key: str | None = None,
```

Comment:
Should this `api_key` be fetched from airflow connection for pinecone?

Reply:
So, it can be passed via the argument or the connection. I added a property to fetch it from either with order of priority being argument and then the connection

Reply:
I agree, that credentials and sensitive information should be a part of the Connections, and better avoid to direct assignment it.

Reply:
@sunank200 , I have removed `api_key` as an arg